tests/test-stbt-docker.sh

```diff
@@ -130,3 +130,18 @@ test_stbt_lint() {
     load_test_pack with-tests
     "$srcdir"/stbt-docker stbt lint --errors-only tests/tests.py
 }
+
+test_stbt_docker_no_new_privs() {
+    [ "$UID" = 1000 ] && [ "$GROUPS" = 1000 ] \
+    ||  skip "Host needs UID/GID 1000 for no-new-privs"
+
+    load_test_pack with-tests
+    DOCKER_OPTS="--security-opt=no-new-privileges" \
+    "$srcdir"/stbt-docker id | grep -v sudo >id || fail "Failed running with no-new-privs"
```

Comment:
Presumably `docker run` automatically adds the user to the `sudo` group if you don't specify `--security-opt=no-new-privileges`?

Reply:
No, that's built-in to `/etc/group` in the image itself. `security-opt=no-new-privileges` sets [`PR_SET_NO_NEW_PRIVS`](https://www.kernel.org/doc/Documentation/prctl/no_new_privs.txt) which disables setuid - so sudo no will no longer run as root.  See also https://www.projectatomic.io/blog/2016/03/no-new-privs-docker/

Reply:
So how come you don't expect to see the word "sudo" in the output of `id`?

Reply:
I see, this is confusing.  Here's what I get if I run `id` via stbt docker.

```
$ DOCKER_OPTS="--security-opt=no-new-privileges" ../../../stbt-docker id
sudo: effective uid is not 0, is /usr/bin/sudo on a file system with the 'nosuid' option set or an NFS file system without root privileges?
uid=1000(stb-tester) gid=1000(stb-tester) groups=1000(stb-tester)
```

The error message is caused by a use of `sudo` in the entrypoint script which this grep is removing.  You're right, if stb-tester were a member of the `sudo` group this test would fail.  As it is the stb-tester is explicitly given sudo rights via `/etc/sudoers.d/stb-tester` rather than via the sudo group so I retract "No, that's built-in to /etc/group in the image itself" that I said earlier, but it is built-in to the image.